src/widgets/list.rs

```diff
@@ -286,7 +286,14 @@ impl<'a> StatefulWidget for List<'a> {
                 } else {
                     (x, list_area.width)
                 };
-                buf.set_line(elem_x, y + j as u16, line, max_element_width);
+                let x_offset = match line.alignment {
+                    Some(Alignment::Center) => {
+                        (area.width / 2).saturating_sub(line.width() as u16 / 2)
+                    }
+                    Some(Alignment::Right) => area.width.saturating_sub(line.width() as u16),
+                    _ => 0,
+                };
+                buf.set_line(elem_x + x_offset, y + j as u16, line, max_element_width);
```

Comment:
I think this may overwrite the bounds as the width isn't changed while the line start position is.
We probably need a few more combinations of tests that cover the edge cases:
- Odd/even for line length and area (to cover the 2 places that do integer division) (4 tests)
- Zero width for both line and area (2 tests)
- Line <, =, > width (3 tests)
- ?

Reply:
I added some tests about this but `test_render_list_alignment_zero_area_width` is panicking for some reason. Can you check?